### DISCUSSION TITLE
Locating custom properties in a iModel

### DISCUSSION THREAD

**Discussion by kardoka** [iTwin/itwinjs-core#5552] [Q&A]
Greetings,

I am transferring this discussion from https://github.com/iTwin/bis-schemas/discussions/209, as I did not receive any feedback there. Do you have any ideas on how this issue could be resolved? If this is not the right place to ask, could you please point me in the right direction?

> We are adding custom properties to our IFC model, but once it is uploaded as a iModel, we can not find these properties via ECSQL queries. We have also tried to locate them via the iTwin Viewer, but they do not seem to appear for any element. In the IFC file, these property sets are defined as `IFCPROPERTYSET` elements.
> 
> The properties can be seen using various IFC viewers, like BIMvision (see green property sets in picture below):
> 
> ![image](https://github-production-user-asset-6210df.s3.amazonaws.com/28572657/238616145-e463565d-08ec-41c6-b92d-3f0cc966e593.png)
> 
> Are custom properties preserved once a IFC model is converted to a iModel? If so, can you provide some general advice on how we should extract these properties via ECSQL queries?

**Comment by grigasp** [comment]
@jffmarker, any ideas?

**Reply by jffmarker** [reply]
@abeesh has already reached out to the IFC Connector team (@prathkk) for more information; we're waiting for a response.

**Reply by prathkk** [reply]
@grigasp  Do we have user IFC file? If yes, could you please share it? Thanks.

**Reply by grigasp** [reply]
I certainly don't. Maybe @kardoka can share?

**Reply by kardoka** [reply]
> I certainly don't. Maybe @kardoka can share?

Yes, I can share. Can I have please have a email address to send it to?

**Reply by prathkk** [reply]
Please send it here. prathamesh.kalelkar@bentley.com

**Reply by prathkk** [reply]
Hi Kardoka,
I replied to this chat few min back but I don't see my answer now. May be I lost it. So replying again. Here is my reply.
I received the HVAC.ifc which you sent. I imported that file to new IModel. I do see all properties correctly. Please see the image below. Could you please import the file once again on your side and verify? Please let me know. Thanks. 
![image](https://github.com/iTwin/itwinjs-core/assets/92881404/20f84866-2809-40fe-89e6-2b5ecdcafde0)

**Reply by kardoka** [reply]
Thank you, I will try this out.
